packages/pds/src/api/com/atproto/handle.ts

```diff
@@ -13,17 +16,25 @@ export default function (server: Server, ctx: AppContext) {
       // self
       did = ctx.cfg.serverDid
     } else {
-      const supportedHandle = ctx.cfg.availableUserDomains.some((host) =>
-        handle.endsWith(host),
-      )
-      if (!supportedHandle) {
-        throw new InvalidRequestError('Not a supported handle domain')
-      }
       const user = await ctx.services.actor(ctx.db).getUser(handle, true)
-      if (!user) {
-        throw new InvalidRequestError('Unable to resolve handle')
+      if (user) {
+        did = user.did
+      } else {
+        const supportedHandle = ctx.cfg.availableUserDomains.some((host) =>
+          handle.endsWith(host),
+        )
+        // this should be in our DB & we couldn't find it, so fail
+        if (supportedHandle) {
+          throw new InvalidRequestError('Unable to resolve handle')
+        }
+
+        // this is not someone on our server, but we help with resolving anyway
+        const resolved = await resolveExternalHandle(ctx.cfg.scheme, handle)
```

Comment:
We talked about doing this: but just want to double check that we are planning to do resolves for users not registered on our server 🤔 

It's relatively inexpensive so I don't see a big downside & shouldnt ever be more than 1 extra hop

Reply:
@pfrazee

Reply:
Yeah I figure do this for now